util.c

```diff
@@ -2379,11 +2390,13 @@ void reopen_mddev(int mdfd)
 	 */
 	char *devnm;
 	int fd;
+
 	devnm = fd2devnm(mdfd);
 	close(mdfd);
 	fd = open_dev(devnm);
 	if (fd >= 0 && fd != mdfd)
 		dup2(fd, mdfd);
+	close_fd(&fd);
```

Comment:
If we assigned same descriptor number as `mdfd` had, we are closing it now - it leads to issues.
it should be only closed in if number is different.

Well, this function is bad, there is no error handling, but looks like it is working.